src/openrct2/ride/Vehicle.cpp

```diff
@@ -7919,7 +7920,7 @@ static void sub_6DBF3E(rct_vehicle* vehicle)
         }
         if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_GO_KART)
         {
-            cx = vehicle->var_CD == 6 ? 18 : 20;
+            cx = vehicle->TrackSubposition == VEHICLE_TRACK_SUBPOSITION_GO_KART_6 ? 18 : 20;
```

Comment:
This is determining the stop positions for gokarts. If in left lane its further along the track than the right lane.

Reply:
Its not possible to overtake in station elements thats why it doesn't check for overtake states (7 and 8)